### PR TITLE
Bookbinder update

### PR DESCRIPTION
**Pull request by guanyilun**
This PR includes several improvements / bug fixes to the existing imprinter + bookbinder stack:

- support passing readout_ids from imprinter to bookbinder and use them in the books 
- allow more parameters in bookbinder to be passed in as arguments
- use G3File instead of smurf_reader in bookbinder
- fix M_index.yaml time units
- several small bugfixes to get bookbinder and imprinter flows working on simons1

### CODE REVIEW

File: sotodlib/io/bookbinder.py [hunk 1]
```diff
@@ -1,19 +1,42 @@
...
+    
+    Returns
+    -------
+    np.ndarray
+        Velocity vector
+
     """
     return np.ediff1d(p)
 
 def smurf_reader(filename):
```
Comment (kmharrington) [inline sotodlib/io/bookbinder.py:27]: We shouldn't use this. This function is identical to `spt3g.core.G3File`.

**Comment by guanyilun** [comment]
OK, I think this is PR at a good place to get merged.

**Comment by kmharrington** [comment]
Hi @guanyilun  could you update your first PR comment with an overview of what's changed in this PR and what to look for in the code?

**Review by kmharrington — CHANGES_REQUESTED** [review]
Ok, I haven't fully understood the details of bookbinder yet but I don't see anything obviously wrong. Please add a summary of the PR (for history's sake) and 2 inline things.

**Review by kmharrington — APPROVED** [review]
thanks! I'm trying to re-run the tests to see if that fixes the errors (seems like a connection thing) if those pass then we're good to merge.

**Pull request merged**